### ISSUE TITLE
Button Component

### ISSUE THREAD

**Issue by nick-thompson** [nick-thompson/blueprint#73]
See #70 

This one's quite simple: just a `<Button>` as part of the core blueprint component library. My gut is to do this just with a `View`, some styles, and an onClick callback, but @JoshMarler had an approach using a Canvas underneath for totally custom drawing which I'm curious about. Let's move that discussion here

**Comment by discordance** [comment]
Following this idea I'm trying to add some fun with:

```
onMouseEnter={this._onMouseEnter}
onMouseLeave={this._onMouseLeave}
onMouseOver={this._onMouseOver}
onMouseOut={this._onMouseOut}
```

Are those ones supported ?

**Comment by nick-thompson** [comment]
Awesome, love the idea :)

They are unsupported currently, but should be quite trivial to add.

See here for the latest work in view events: https://github.com/nick-thompson/blueprint/pull/95/files

With juce we should be able to just tap into the extra mouse events on [juce::Component](https://docs.juce.com/master/classComponent.html). I'm working on extending the View event logic to be closer to the web Event and UIEvent spec– I'll add that to the PR shortly and I'll throw in these extra events while I'm at it

**Comment by JoshMarler** [comment]
@nick-thompson , Good to close this now ?

**Comment by nick-thompson** [comment]
Yup! Merged in #118
